### PR TITLE
support multiple repl options.

### PR DESCRIPTION
**Pull request by roachsinai**
For #57.

### CODE REVIEW

**Comment by roachsinai** [comment]
With setting:

```
let g:repl_program = {
            \   'python': ['/home/roach/.pyenv/versions/3.7.4/envs/nn/bin/ipython',
            \              'optirun /home/roach/.pyenv/versions/3.7.4/envs/nn/bin/ipython'],
}
```

Get result below:

image

File: autoload/repl.vim [hunk 1]
```diff
@@ -73,7 +73,20 @@ function! repl#REPLGetName()
     elseif &buftype ==# 'terminal'
 		return bufname('%')[1:]
 	elseif has_key(g:repl_program, &filetype)
-		return g:repl_program[&filetype]
```
Comment (roachsinai) [inline autoload/repl.vim:76]: Origin file use <kbd>tab</kbd>, so the indent is not right on github changes.

**Comment by sillybun** [comment]
Great job!

**Comment by sillybun** [comment]
Currently, the old setting is invalid:

```
let g:repl_program = {
            \   'python': 'ipython',
            \   'default': 'zsh',
            \   'r': 'R',
            \   'lua': 'lua',
            \   'vim': 'vim -e',
            \   }
```

I think we should support both the list and string configuration.

```
let g:repl_program = {
            \   'python': ['ipython','python'],
            \   'default': ['zsh'],
            \   'r': ['R'],
            \   'lua': ['lua'],
            \   'vim': ['vim -e']
            \   }
```

**Comment by sillybun** [comment]
I have fixed the problem. Now you can write it both in list version or string version.

**Pull request merged**